vector/src/main/java/im/vector/app/features/settings/VectorSettingsPreferencesFragment.kt

```diff
@@ -71,6 +71,14 @@ class VectorSettingsPreferencesFragment @Inject constructor(
             }
         }
 
+        findPreference<VectorSwitchPreference>(VectorPreferences.SETTINGS_SHOW_PRESENCE)!!.let { pref ->
+            pref.isChecked = vectorPreferences.showPresence()
+            pref.setOnPreferenceChangeListener { _, _ ->
+                MainActivity.restartApp(requireActivity(), MainActivityArgs(clearCache = false))
```

Comment:
I wonder if we can do better than restarting the MainActivity to take changes into account

Reply:
Why do we need an activity restart here ? vectorPreferences.showPresence() should have the updated value after user changes the setting, it doesn't ?

Reply:
If the problem is for example the room list recyclerview, maybe you can try to call `requestModelBuild`  to request a model update so that epoxy will rebuild the UI for the current updated data